theano/gof/type.py

```diff
@@ -795,3 +796,137 @@ def signature(self):
         return (self.type,)
 
 CDataType.Constant = CDataTypeConstant
+
+
+class EnumType(Type, dict):
```

Comment:
The name is EnumType, but this don't create or represent an enum, but macros. I'm not sure what to do about that, so at least, this should be in the docstring.

I'm not sure if we could reuse the enum from cudnn. Could CData type help us here? Or create something like that for enum? This way we will reuse the original enum and that remove the conversion in C. Right now, I think this PR move the conversion from Python to C. With the master of Theano, we can move the get_op_param to params if we hardcode the enum values that correspond to the cuda enum.